tests/unit/common/test_alignedtable.py

```diff
@@ -189,6 +189,26 @@ def test_init_with_duplicate_custom_categories(self):
                 description='Test aligned container',
                 category_tables=categories)
 
+    def test_init_with_bad_custom_categories(self):
+        """Test that we cannot provide a category that is not a DynamicTable"""
+        num_rows = 10
+        categories = [  # good category
+                      DynamicTable(name='test1',
+                                   description="test1 description",
+                                   columns=[VectorData(name='test1'+t,
+                                                       description='test1' + t + ' description',
+                                                       data=np.arange(num_rows)) for t in ['c1', 'c2', 'c3']]
+                                   ),
+                      # use a list as a bad category example
+                      [0, 1, 2]]
+        with self.assertRaises(ValueError) as ve:
+            AlignedDynamicTable(
+                name='test_aligned_table',
+                description='Test aligned container',
+                category_tables=categories)
+            self.assertEqual(str(ve.exception),
+                             "Category table with index 1 is not a DynamicTable")
```

Comment:
hdmf.testing.TestCase provides `assertRaisesWith(exception, str)` which simplifies this statement. This form is used throughout hdmf tests, so I would recommend changing it to the following:

```suggestion
        msg = "Category table with index 1 is not a DynamicTable"
        with self.assertRaisesWith(ValueError, msg):
            AlignedDynamicTable(
                name='test_aligned_table',
                description='Test aligned container',
                category_tables=categories)
```

Reply:
Sounds good. I updated the tests in test_alignedtable.py to use asertRaisesWith throughout (i.e., aside from one case where doing a full match of the error message is too messy).